### PR TITLE
Ensured volume mounts are sorted consistently

### PR DESCRIPTION
**Pull request by chatton**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

This PR ensures when sorting volume mounts all of the fields are taken into consideration.

### CODE REVIEW

File: pkg/util/merge/merge_test.go [hunk 1]
```diff
@@ -588,20 +585,6 @@ func TestMergeVolumeAddVolume(t *testing.T) {
 	assert.Equal(t, corev1.EmptyDirVolumeSource{}, *volume1.EmptyDir)
 }
 
-func TestMergeVolumeMounts(t *testing.T) {
```
Comment (chatton) [inline pkg/util/merge/merge_test.go:591]: this is actually the incorrect merging behaviour - uniqueness should be determined by name, path and subpath

File: pkg/util/merge/merge_test.go [hunk 2]
```diff
@@ -239,7 +239,7 @@ func TestMergeContainer(t *testing.T) {
 
 		t.Run("Volume Mounts are overridden", func(t *testing.T) {
 			volumeMounts := mergedContainer.VolumeMounts
-			assert.Len(t, volumeMounts, 2)
+			assert.Len(t, volumeMounts, 3, "volume mounts can have the same name, the uniqness is the combination of name, path and subpath")
```
Comment (bznein) [inline pkg/util/merge/merge_test.go:242]:
```suggestion
			assert.Len(t, volumeMounts, 3, "volume mounts can have the same name, the uniqueness is the combination of name, path and subpath")
```

**Review by bznein — APPROVED** [review]

File: pkg/util/merge/merge.go [hunk 3]
```diff
@@ -430,16 +431,21 @@ func VolumeMounts(original, override []corev1.VolumeMount) []corev1.VolumeMount
...
-		return mergedMounts[i].Name < mergedMounts[j].Name
+		return volumeMountToString(mergedMounts[i]) < volumeMountToString(mergedMounts[j])
 	})
 
 	return mergedMounts
 }
 
+// volumeMountToString returns a string consisting of all components of the given VolumeMount.
+func volumeMountToString(v corev1.VolumeMount) string {
+	return strings.Join([]string{v.Name, v.MountPath, v.SubPath}, "-")
```
Comment (alyacb) [inline pkg/util/merge/merge.go:442]: [q] Is it guaranteed that none of `Name`, `MountPath`, and `SubPath` will contain the "-" character? The reason I ask is so we don't get a case where we have two different (valid) volume mounts with the same `volumeMountToString()` value, e.g.:
```
Name: a-b
MountPath: c
SubPath: d
```
and 
```
Name: a
MountPath: b-c
SubPath: d
```
Will both have volumeMountToString: `a-b-c-d`

File: pkg/util/merge/merge_test.go (hunk 2 above)
Comment (alyacb) [inline pkg/util/merge/merge_test.go]: [nit] Should it be "uniqueness", not "uniqness"?

**Review by alyacb — COMMENTED** [review]
Just a question and a nit, otherwise looks good!

File: pkg/util/merge/merge_test.go (hunk 2 above)
Comment (alyacb) [reply pkg/util/merge/merge_test.go]: Resolving! You fixed it before I could submit the comment 🎉

File: pkg/util/merge/merge.go (hunk 3 above)
Comment (chatton) [reply pkg/util/merge/merge.go:442]: good point! Maybe we can change it to include a character that will not be able to appear in the names/paths

File: pkg/util/merge/merge_test.go (hunk 2 above)
Comment (chatton) [reply pkg/util/merge/merge_test.go]: yes! Addressed in this commit https://github.com/mongodb/mongodb-kubernetes-operator/pull/300/commits/b2896a20a2281c621b1927209bfc73ec36d2ea6a

**Review by alyacb — APPROVED** [review]
LGTM! 🎉

**Pull request merged**